book/src/pong-tutorial/pong-tutorial-04.md

```diff
@@ -0,0 +1,422 @@
+previous chapters to add a new object to our game: a ball that moves
+and bounces around!
+
+First, let's define some other useful constants for this chapter:
+```rust,no_run,noplaypen
+pub const BALL_VELOCITY_X: f32 = 75.0;
+pub const BALL_VELOCITY_Y: f32 = 50.0;
+pub const BALL_RADIUS: f32 = 2.0;
+```
+
+## Load the ball sprite
+
+In [a previous chapter][pong_02_drawing] we saw how to load a sprite sheet
+and get things drawn on the screen. In this section we're going to
+add the ball.
+
+In `pong.rs` file, modify the `load_sprite_sheet` function using the
+following lines of code.
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# use amethyst::prelude::*;
+# use amethyst::assets::{Loader, AssetStorage};
+# use amethyst::renderer::{Texture, PngFormat, TextureHandle, MaterialTextureSet,
+#                          TextureCoordinates, Sprite, SpriteSheet, SpriteSheetHandle, TextureMetadata};
+# use amethyst::ecs::World;
+# const PADDLE_HEIGHT: f32 = 16.0;
+# const PADDLE_WIDTH: f32 = 4.0;
+# const SPRITESHEET_SIZE: (f32, f32) = (8.0, 16.0);
+# const BALL_RADIUS: f32 = 2.0;
+# fn load_sprite_sheet(world: &mut World) -> SpriteSheetHandle {
+#   let texture_handle = {
+#       let loader = world.read_resource::<Loader>();
+#       let texture_storage = world.read_resource::<AssetStorage<Texture>>();
+#       loader.load(
+#           "texture/pong_spritesheet.png",
+#           PngFormat,
+#           TextureMetadata::srgb_scale(),
+#           (),
+#           &texture_storage,
+#       )
+#   };
+#   let texture_id = 0;
+#   let mut material_texture_set = world.write_resource::<MaterialTextureSet>();
+#   material_texture_set.insert(texture_id, texture_handle);
+# let tex_coords = TextureCoordinates {
+#    left: 0.0,
+#    right: PADDLE_WIDTH / SPRITESHEET_SIZE.0,
+#    bottom: 1.0 - PADDLE_HEIGHT / SPRITESHEET_SIZE.1,
+#    top: 1.0,
+# };
+# let paddle_sprite = Sprite {
+#    width: PADDLE_WIDTH,
+#    height: PADDLE_HEIGHT,
+#    offsets: [PADDLE_WIDTH / 2.0, PADDLE_HEIGHT / 2.0],
+#    tex_coords
+# };
+// Create the sprite for the ball.
+let ball_diameter = BALL_RADIUS * 2.0;
+let tex_coords = TextureCoordinates {
+    left: PADDLE_WIDTH / SPRITESHEET_SIZE.0,
+    right: 1.0,
+    bottom: 1.0 - ball_diameter / SPRITESHEET_SIZE.1,
+    top: 1.0,
+};
+let ball_sprite = Sprite {
+    width: ball_diameter,
+    height: ball_diameter,
+    offsets: [BALL_RADIUS, BALL_RADIUS],
+    tex_coords,
+};
+
+// Collate the sprite layout information into a sprite sheet
+let sprite_sheet = SpriteSheet {
+    texture_id,
+    sprites: vec![paddle_sprite, ball_sprite], // <- update this line to add 'ball_sprite'
+};
+# let sprite_sheet_handle = {
+#    let loader = world.read_resource::<Loader>();
+#    let sprite_sheet_store = world.read_resource::<AssetStorage<SpriteSheet>>();
+#    loader.load_from_data(sprite_sheet, (), &sprite_sheet_store)
+# };
+#  sprite_sheet_handle
+# }
+```
+
+## Create our next Component: The ball Conponent!
```

Comment:
`Create our next component: The ball component!` <- there's a `n` instead of a `m`